### PR TITLE
add transaction creator check

### PR DESCRIPTION
**Pull request by Solonets**
Signed-off-by: Moonraker <ssolonets@gmail.com>

### Description of the Change

Adds check whether a user can query a transaction
https://soramitsu.atlassian.net/browse/IR-1039

### Benefits

Don't return transaction which user doesn't have access to

### Possible Drawbacks 

-

### CODE REVIEW

File: irohad/model/query_execution.hpp [hunk 1]
```diff
@@ -125,7 +125,8 @@ namespace iroha {
           const shared_model::interface::GetAccountTransactions &query);
 
       QueryResponseBuilderDone executeGetTransactions(
-          const shared_model::interface::GetTransactions &query);
+          const shared_model::interface::Query &query,
```
Comment (l4l) [inline irohad/model/query_execution.hpp]: Strange parameter in signature

File: irohad/model/impl/query_execution.cpp [hunk 2]
```diff
@@ -313,17 +313,21 @@ QueryProcessingFactory::executeGetAccountTransactions(
...
   std::vector<shared_model::proto::Transaction> txs;
+  bool can_get_all = checkAccountRolePermission(
+      query.creatorAccountId(), *_wsvQuery, can_get_all_txs);
   transactions.subscribe([&](const auto &tx) {
     if (tx) {
-      txs.push_back(
-          *std::static_pointer_cast<shared_model::proto::Transaction>(*tx));
+      auto proto_tx = *std::static_pointer_cast<shared_model::proto::Transaction>(*tx);
+      if (can_get_all or proto_tx.creatorAccountId() == query.creatorAccountId()  )
+        txs.push_back(proto_tx);
```
Comment (l4l) [inline irohad/model/impl/query_execution.cpp]: I'm not sure it is an issue, but prefer use `clone` explicitly (cloneable.hpp) and declare var with ref aka `auto &proto_tx`

File: test/integration/acceptance/get_transactions_test.cpp [hunk 3]
```diff
@@ -221,7 +221,7 @@ TEST_F(GetTransactions, InexistentHash) {
  * @then TransactionsResponse with no transactions
  * TODO(@l4l) 02/01/18 Should be enabled after resolving IR-1039
```
Comment (l4l) [inline test/integration/acceptance/get_transactions_test.cpp]: remove todo plz

File: irohad/model/impl/query_execution.cpp [hunk 4]
```diff
@@ -313,17 +313,21 @@ QueryProcessingFactory::executeGetAccountTransactions(
...
+  bool can_get_all =
+      checkAccountRolePermission(accountId, *_wsvQuery, can_get_all_txs);
   transactions.subscribe([&](const auto &tx) {
     if (tx) {
-      txs.push_back(
-          *std::static_pointer_cast<shared_model::proto::Transaction>(*tx));
+      auto proto_tx =
+          *std::static_pointer_cast<shared_model::proto::Transaction>(*tx);
+      if (can_get_all or proto_tx.creatorAccountId() == accountId)
+        txs.push_back(proto_tx);
```
Comment (l4l) [inline irohad/model/impl/query_execution.cpp]: Still concerned about that line, ping @lebdron

Comment (lebdron) [reply irohad/model/impl/query_execution.cpp]: `proto_tx` is pushed with lvalue, so copy ctor is called, looks fine to me.

**Pull request merged**